sources/packages/web/src/components/common/OfferingForm.vue

```diff
@@ -1,32 +1,70 @@
 <template>
-  <!--TODO: Use formio-container component here once LocationProgramOffering.vue
-  is refactored. -->
-  <formio
+  <formio-container
     formName="educationprogramoffering"
-    :data="data"
+    :formData="formData"
     @submitted="saveOffering"
-  ></formio>
+  >
+    <template #actions="{ submit }" v-if="!readOnly">
+      <footer-buttons
+        :processing="processing"
+        primaryLabel="Submit"
+        @primaryClick="submit"
+        :showSecondaryButton="false"
+      />
+    </template>
+  </formio-container>
 </template>
 
 <script lang="ts">
-import { OfferingDTO } from "@/types";
-import { SetupContext } from "vue";
+import { FormIOForm, OfferingFormModel } from "@/types";
+import { SetupContext, computed } from "vue";
+import { useOffering } from "@/composables";
+import { AuthService } from "@/services/AuthService";
 
 export default {
   props: {
     data: {
       type: Object,
       required: true,
+      default: {} as OfferingFormModel,
+    },
+    //For view only purpose where submit action not required.
+    readOnly: {
+      type: Boolean,
+      required: false,
+      default: true,
     },
   },
   emits: ["saveOffering"],
-  setup(_props: any, context: SetupContext) {
-    const saveOffering = (data: OfferingDTO) => {
-      context.emit("saveOffering", data);
+  setup(props: any, context: SetupContext) {
+    const { mapOfferingChipStatus } = useOffering();
+    const saveOffering = (form: FormIOForm) => {
+      context.emit("saveOffering", form.data);
     };
+    /**
+     * The property clientType is populated for institution because
+     * the form.io for education program offering has a logic at it's root level panel
+     * to disable all the form inputs when clientType is not institution.
+     * The above mentioned logic is added to the panel of the form to display the
+     * form as read-only for ministry(AEST) user and also allow the hidden component values
+     * to be calculated.
+     *! If a form.io is loaded with readOnly attribute set to true, then the restricts
+     *! hidden components to calculate it's value by design.
+     */
+    const formData = computed(
+      (): OfferingFormModel => ({
+        ...props.data,
+        offeringChipStatus: props.data.offeringStatus
```

Comment:
I believe that this could be just as below:
````typescript
offeringChipStatus: mapOfferingChipStatus(props.data.offeringStatus),
````

Reply:
The offering form needs a undefined|null value to chip status during create new offering. 
Otherwise it is loading status chip with default value which is incorrect.